pkg/mutation/mutation_function.go

```diff
@@ -171,36 +182,15 @@ func createMissingElement(m types.Mutator, current interface{}, previous interfa
 		next = make([]interface{}, 0)
 	}
 
-	// Append to element of type
-	switch castPathEntry := pathEntry.(type) {
-	case *parser.Object:
-		currentAsObject, ok := current.(map[string]interface{})
-		if !ok { // Path entry type does not match current object
-			return fmt.Errorf("mismatch between path entry (type: object) and received object (type: %T). Path: %+v", current, castPathEntry)
-		}
-		currentAsObject[castPathEntry.Reference] = next
-	case *parser.List:
-		currentAsList, ok := current.([]interface{})
-		if !ok { // Path entry type does not match current object
-			return fmt.Errorf("mismatch between path entry (type: List) and received object (type: %T). Path: %+v", current, castPathEntry)
-		}
-		current = append(currentAsList, next)
+	// Set new keyfield
+	if castPathEntry, ok := pathEntry.(*parser.List); ok {
 		nextAsObject, ok := next.(map[string]interface{})
 		if !ok { // Path entry type does not match current object
-			return fmt.Errorf("two consecutive list path entries not allowed: %+v %+v", castPathEntry, nextPathEntry)
+			return nil, fmt.Errorf("two consecutive list path entries not allowed: %+v %+v", castPathEntry, nextPathEntry)
 		}
 		nextAsObject[castPathEntry.KeyField] = *castPathEntry.KeyValue
```

Comment:
Is this dereference safe?

Reply:
It should be... I don't think it's possible to trigger a create against a globbed path entry as a missing entry just doesn't fan out.

We can return an error per above, if you prefer?

Reply:
Yes, I'd prefer not depending on too many far-away assumptions to avoid panics, as they have a habit of changing under your feet. And even when they don't, it makes it harder to reason about correctness of the code in isolation.

Reply:
+1

Reply:
added nil checking